runconfig/parse.go

```diff
@@ -187,6 +188,13 @@ func Parse(cmd *flag.FlagSet, args []string) (*Config, *HostConfig, *flag.FlagSe
 			MemorySwap = parsedMemorySwap
 		}
 	}
+	if *flSwappiness < 0 || *flSwappiness > 100 {
+		return nil, nil, cmd, fmt.Errorf("invalid value:%d. valid memory swappiness range is 0-100", *flSwappiness)
+	}
+	// This is to ensure that we do not try to override swappiness value for default case
+	if *flSwappiness == 60 {
+		*flSwappiness = -1
+	}
```

Comment:
Yes, there would be problem if we using it on kernel older than 3.16 with use_hierarchy=1, but I remember there is another PR handling this, no?
Probably we should ignore any value set to swappiness in this case, or just throw error out. If you want to handle specific kernel version, might pick up this commit: https://github.com/hqhq/docker/commit/16a0e0ec51c092eb28c3f4e96ff70afbfb879b66